storm-core/src/jvm/org/apache/storm/daemon/supervisor/SyncProcessEvent.java

```diff
@@ -0,0 +1,427 @@
+        data.put(Config.TOPOLOGY_SUBMITTER_USER, user);
+        data.put("worker-id", workerId);
+
+        Set<String> logsGroups = new HashSet<>();
+        //for supervisor-test
+        if (stormconf.get(Config.LOGS_GROUPS) != null) {
+            List<String> groups = (List<String>) stormconf.get(Config.LOGS_GROUPS);
+            for (String group : groups){
+                logsGroups.add(group);
+            }
+        }
+        if (stormconf.get(Config.TOPOLOGY_GROUPS) != null) {
+            List<String> topGroups = (List<String>) stormconf.get(Config.TOPOLOGY_GROUPS);
+            logsGroups.addAll(topGroups);
+        }
+        data.put(Config.LOGS_GROUPS, logsGroups.toArray());
+
+        Set<String> logsUsers = new HashSet<>();
+        if (stormconf.get(Config.LOGS_USERS) != null) {
+            List<String> logUsers = (List<String>) stormconf.get(Config.LOGS_USERS);
+            for (String logUser : logUsers){
+                logsUsers.add(logUser);
+            }
+        }
+        if (stormconf.get(Config.TOPOLOGY_USERS) != null) {
+            List<String> topUsers = (List<String>) stormconf.get(Config.TOPOLOGY_USERS);
+            for (String logUser : topUsers){
+                logsUsers.add(logUser);
+            }
+        }
+        data.put(Config.LOGS_USERS, logsUsers.toArray());
+        writeLogMetadataToYamlFile(stormId, port, data, conf);
+    }
+
+    /**
+     * run worker as user needs the directory to have special permissions or it is insecure
+     * 
+     * @param stormId
+     * @param port
+     * @param data
+     * @param conf
+     * @throws IOException
+     */
+    public void writeLogMetadataToYamlFile(String stormId, Long port, Map data, Map conf) throws IOException {
+        File file = ConfigUtils.getLogMetaDataFile(conf, stormId, port.intValue());
+
+        if (!Utils.checkFileExists(file.getParent())) {
+            if (Utils.getBoolean(conf.get(Config.SUPERVISOR_RUN_WORKER_AS_USER), false)) {
+                FileUtils.forceMkdir(file.getParentFile());
+                SupervisorUtils.setupStormCodeDir(conf, ConfigUtils.readSupervisorStormConf(conf, stormId), file.getParentFile().getCanonicalPath());
+            } else {
+                file.getParentFile().mkdirs();
+            }
+        }
+        FileWriter writer = new FileWriter(file);
+        Yaml yaml = new Yaml();
+        try {
+            yaml.dump(data, writer);
+        }finally {
+            writer.close();
+        }
+    }
+
+    /**
+     * Create a symlink from workder directory to its port artifacts directory
+     * 
+     * @param conf
+     * @param stormId
+     * @param port
+     * @param workerId
+     */
+    protected void createArtifactsLink(Map conf, String stormId, Long port, String workerId) throws IOException {
+        String workerDir = ConfigUtils.workerRoot(conf, workerId);
+        String topoDir = ConfigUtils.workerArtifactsRoot(conf, stormId);
+        if (Utils.checkFileExists(workerDir)) {
```

Comment:
Minor there was a log message here, would be nice to at least still have a debug message for it.

```
(log-message "Creating symlinks for worker-id: " worker-id " storm-id: " storm-id " to its port artifacts directory")
```